### PR TITLE
Model Training Implemented

### PR DESCRIPTION
**Pull request by apoorvjha**
Implemented the code to appropriately choose the loss function based on the type of network (Viz., Actor or Critic) and apply the optimization appropriately. The choice of device for model training viz CPU and GPU is also integrated.

### CODE REVIEW

**Review by apoorvjha — COMMENTED** [review]
Fix the Actor network loss function, It should be maximizing the mean of the policy gradient values rather than minimizing.
